### PR TITLE
Implement --opbeans-java-agent-branch option

### PR DESCRIPTION
**Pull request by felixbarny**
This lets you start Opbeans with a specific Java agent branch built from source

It adds very little overhead when not specifying a branch

### CODE REVIEW

File: docker/opbeans/java/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
+RUN mkdir /agent \
+  # making sure there is at least one file to COPY (otherwise docker complains)
+  && touch /agent/ignore
+COPY build-agent.sh .
+# noop if JAVA_AGENT_BRANCH is not set
+RUN ./build-agent.sh "${JAVA_AGENT_REPO}" "${JAVA_AGENT_BRANCH}"
 
+FROM opbeans/opbeans-java:latest
+# replaces the /agent/elastc-apm-agent.jar if it has been built by build-agent.sh
+COPY --from=0 /agent/* /app
```
Comment (graphaelli) [inline docker/opbeans/java/Dockerfile]: This caused an error during build for me:

```
Step 8/11 : COPY --from=0 /agent/* /app
ERROR: Service 'opbeans-java' failed to build: When using COPY with more than one source file, the destination must be a directory and end with a /
```

File: docker/opbeans/java/Dockerfile [hunk 2]
```diff
@@ -1,10 +1,24 @@
-FROM opbeans/opbeans-java:latest
+FROM maven:3.5.3-jdk-10
+# setting --build-arg JAVA_AGENT_BRANCH=<branch> causes the agent to be built from source
+# instead of using the agent which comes pre-built with opbeans-java:latest
+ARG JAVA_AGENT_REPO=elastic/apm-agent-java
+ARG JAVA_AGENT_BRANCH=
```
Comment (graphaelli) [inline docker/opbeans/java/Dockerfile]: This is currently always set by `compose.py`, I think changing `"master"` to `""` should be good enough.

https://github.com/elastic/apm-integration-testing/blob/b0cef3414173aa3e5ec8100d1e2ec2b0d810f504/scripts/compose.py#L1730

File: docker/opbeans/java/Dockerfile (hunk 1 above)
Comment (felixbarny) [reply docker/opbeans/java/Dockerfile]: seems to depend on the docker version. it works on my machine. Suffixed `/app/` with a slash, that should work in both cases.

File: docker/opbeans/java/Dockerfile (hunk 2 above)
Comment (felixbarny) [reply docker/opbeans/java/Dockerfile]: done

**Comment by v1v** [comment]
jenkins run the tests please

**Comment by felixbarny** [comment]
I have no idea why this fails 🙇

**Comment by v1v** [comment]
@felixbarny  I took the chance to fix the test with https://github.com/elastic/apm-integration-testing/pull/472/commits/e4e2eab5fe8ee9915586f9bd143c7944271734de 

The issue was when running `make test` which caused the UTs to fail as the default value was removed.

**Comment by felixbarny** [comment]
> I'd rather enforce the master branch within the script as done previously for backward compatiblity

But that means the agent will always be built from source 🤔

**Comment by v1v** [comment]
That's indeed a fact! I'd then revert that change and fix the UTs if you don't mind

**Comment by felixbarny** [comment]
Tests are green!
@v1v as soon as you approve we can merge this I think

**Pull request merged**